Tests for encoding of macro parameters

mrosanes opened the pull request:
Insert tests to check the correct encoding/decoding of macro
parameters.

Using insertTest decorator, different parameter combinations
(including repeat parameters) are tested.

Commit:
Tests for encoding of macro parameters

Insert tests to check the correct encoding/decoding of macro
parameters.

Using insertTest decorator, different parameter combinations
(including repeat parameters) are tested.

Commit:
Add tests

reszelaz commented on src/sardana/taurus/core/tango/sardana/test/res/params_definitions.py:
```diff
@@ -0,0 +1,119 @@
+
+################################################################################
+
+params_definition_move = [
```
I would reference in the variable names with the macro examples located in `sardana.macroserver.macros.examples.parameters`. So `pt0_param_def`, `pt01_param_def`, ...

reszelaz replied:
Apart of that I would make a dedicated module called: `paramdef` where I would place all these definitions together with the one already defined in `test_macro` module.

mrosanes replied:
Done

reszelaz commented on src/sardana/taurus/core/tango/sardana/test/res/params_definitions.py:
```diff
@@ -0,0 +1,119 @@
+    ]
+
+
+move_parameters_1 = [[["mot01", "0"]]]
```
Personally I would keep the parameter values close to the `insertTest` calls so in the test module. I think that we will gain the readability. But treat is just as a suggestion.

mrosanes replied:
Done

reszelaz commented on src/sardana/taurus/core/tango/sardana/test/test_macro_paramrepeat.py:
```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
```
I would merge this module with the already existing one `test_macro`.

mrosanes replied:
Done

Commit:
Add tests and review code

- Add tests for new cases.
- Rename resource file.
- Move input and expected parameters to test file.
- Rename param definitions according to macro examples convention.

Commit:
Merge parameter related tests in a single file

Merge parameter related tests in a single file: tests referring
to: sardana.taurus.core.tango.sardana.test.macro.py

Rename parameter definitions used as resources for tests.

Add documentation.

Commit:
Move param definitions to util module

Move param definitions to util module, instead to a resource
module.

Commit:
Modify import style

Modify import style for util module to adapt it to Sardana
code conventions.

reszelaz committed:
Fix fromList of SingleParamNode

Fix fromList of SingleParamNode to accept brackets for param repets with
just one member.

reszelaz committed:
Fix fromList of RepeatNode

Change algorithm to iterate over param_def instead over param values.

reszelaz committed:
Fix tests

Correct some expected values and remove some tests which are not necessary.

Commit:
Unify name for MacroNode name

Unify name for MacroNode. Use a default macro name for all
tests.

Commit:
Do not allow brackets for single elements

Simplify the interface by not allowing brackets on
param repeats, where the repetitions are formed by
individual elements repeating.

In the other hand, the brackets are mandatory for
param repeats where the repetitions have two or
more elements for each repetition.

Commit:
Adding tests for new param definition

Adding tests for param definition containing a param repeat of
a single element (a list of numbers), followed by a single
parameter ('motor name').

Commit:
Add tests for nested parameters

Add tests for testing input of nested param repeats.

Commit:
Corrected tests of nested param repeats

Correct and update tests of nested param repeats.
